lib/generics/threadsafemap.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generics
```

Comment:
I don't think this is necessary here, I'd personally stick to a simple `map[string]string` protected by a mutex.

Reply:
I have to agree here. This feels largely unneeded. Mutexes for maps are often grouped to protect more than one field in the vast majority of cases in the Teleport codebase IMO.